utils/debug/Debug.js

```diff
@@ -0,0 +1,128 @@
+    const nbObjectsChartLabel = ['0s'];
+    const nbObjectsChart = new Chart('nb-objects', {
+        type: 'line',
+        data: {
+            labels: nbObjectsChartLabel,
+            datasets: [nbObjectsDataset],
+        },
+        options: {
+            animation: { duration: 10 },
+            scales: {
+                yAxes: [{
+                    display: true,
+                    ticks: {
+                        suggestedMin: 0, // minimum will be 0, unless there is a lower value.
+                    },
+                }],
+            },
+        },
+    });
+
+    function updateNbObjectsChart() {
+        function countElem(node) {
+            if (!node) {
+                return 0;
+            }
+            let count = 1; // this node
+            if (node.children) {
+                for (const child of node.children) {
+                    count += countElem(child);
+                }
+            }
+            return count;
+        }
+
+        const timeInS = Math.floor((Date.now() - timestamp) / 1000);
+
+        const limit = 50;
+        nbObjectsChartLabel.push(`${timeInS}s`);
+        if (nbObjectsChartLabel.length > limit) {
+            nbObjectsChartLabel.shift();
+        }
+        nbObjectsDataset.data.push({ x: timeInS, y: countElem(scene.gfxEngine.scene3D) });
+        if (nbObjectsDataset.data.length > limit) {
+            nbObjectsDataset.data.shift();
+        }
+        nbObjectsChart.update();
+    }
+
+    // hook that to scene.update
+    const oldUpdate = Object.getPrototypeOf(scene).update.bind(scene);
+    scene.update = function debugUpdate() {
+        oldUpdate();
+        updateNbObjectsChart();
+    };
+
+    // DEBUG CONTROLS
+    const gui = menuGlobe.gui.addFolder('Debug Tools');
+
+    const state = {
+        showOutline: false,
+        wireframe: false,
+        displayCharts: false,
+    };
+
+    // charts
+    gui.add(state, 'displayCharts').name('Display charts').onChange((newValue) => {
+        if (newValue) {
+            chartDiv.style.display = 'flex';
+        } else {
+            chartDiv.style.display = 'none';
+        }
+    });
+
+    function applyToNodeFirstMaterial(cb) {
+        scene.getMap().tiles.children[0].traverse((object) => {
+            if (object.materials) {
+                cb(object.materials[0]);
+            }
+        });
+        scene.renderScene3D();
+    }
+
+    // tiles outline
+    gui.add(state, 'showOutline').name('Show tiles outline').onChange((newValue) => {
+        applyToNodeFirstMaterial((material) => {
+            Object.getPrototypeOf(material)._showOutline = newValue;
+            material.uniforms.showOutline.value = newValue;
+        });
+    });
+
+    // tiles wireframe
+    gui.add(state, 'wireframe').name('Wireframe').onChange((newValue) => {
+        applyToNodeFirstMaterial((material) => {
+            Object.getPrototypeOf(material)._wireframe = newValue;
+            material.wireframe = newValue;
```

Comment:
- Somes users can use wireframe for study alignment, occlusion, arrangement (use cartography architecture,  and debug). Many applications offer wireframe mode without using a debug mode, because they have a different purpose than debugging. 
- Somes users can use outline, it's a classic mapping tool ([example](http://mondecarte.o.m.f.unblog.fr/files/2014/11/carte-du-monde-9.jpg))
- Modifying prototypes does not allow wireframe by entity
- The option wireframe must be placed in BasicMaterial